spec/indexers/hyrax/valkyrie_work_indexer_spec.rb

```diff
@@ -9,23 +9,22 @@
   it_behaves_like 'a Work indexer'
 
   context 'when extending with basic metadata' do
-    before do
-      module Hyrax::Test
-        module Basic
-          class Work < Hyrax::Work
-            include Hyrax::Schema(:basic_metadata)
-          end
-        end
-      end
-    end
-    after { Hyrax::Test.send(:remove_const, :Basic) }
-
     let(:indexer_class) do
       Class.new(described_class) do
         include Hyrax::Indexer(:basic_metadata)
       end
     end
-    let(:resource) { Hyrax::Test::Basic::Work.new }
+    let(:resource) do
+      Class.new(Hyrax::Work) do
+        # Included to address a `ArgumentError: Class name cannot be
+        # blank. You need to supply a name argument when anonymous
+        # class given`"
+        def self.model_name
+          ActiveModel::Name.new(self, nil, "TemporaryResource")
+        end
```

Comment:
@no-reply Lines 22-24 represent the change from master to address the ArgumentError.  Does this change fit with your expectation for code resiliency?